x/perp/v2/keeper/dnr_test.go

```diff
@@ -97,3 +97,110 @@ func TestUserVolumes(t *testing.T) {
 	}
 	NewTestSuite(t).WithTestCases(tests...).Run()
 }
+
+func TestDiscount(t *testing.T) {
+	alice := testutil.AccAddress()
+	pairBtcNusd := asset.Registry.Pair(denoms.BTC, denoms.NUSD)
+	positionSize := sdk.NewInt(10_000)
+	startBlockTime := time.Now()
+
+	exchangeFee := sdk.MustNewDecFromStr("0.0010")           // 0.1%, default fee taken from CreateCustomMarketAction
+	globalFeeDiscount := sdk.MustNewDecFromStr("0.0005")     // 0.05%
+	fauxGlobalFeeDiscount := sdk.MustNewDecFromStr("0.0006") // 0.06%
+	customFeeDiscount := sdk.MustNewDecFromStr("0.0002")     // 0.02%
+	fauxCustomFeeDiscount := sdk.MustNewDecFromStr("0.0003") // 0.03%
+
+	tests := TestCases{
+		TC("user does not have any past epoch volume: no discount applies").
+			Given(
+				DnREpochIs(1),
```

Comment:
Maybe this is not needed since it is included in the When?

Reply:
addressed!